packages/xo-server/src/api/vm.js

```
@@ -746,7 +746,9 @@ export async function snapshot ({
 }) {
   await checkPermissionOnSrs.call(this, vm)
 
-  return (await this.getXapi(vm).snapshotVm(vm._xapiRef, name)).$id
+  const snapshotId = (await this.getXapi(vm).snapshotVm(vm._xapiRef, name)).$id
+  await this.addAcl(this.user.id, snapshotId, 'admin')
```

What if this fail?

---

Either remove snapshot on failure or ignore the error.